### PR TITLE
Show an elaboration when accessing a non-existent property of a union type

### PR DESCRIPTION
**Pull request by oijazsh**
This fixes issue #10256 by showing an elaboration when accessing a non-existant property of a union type. It does not show an elaboration in the following cases:

* If the union is a enum
* If all the subtypes of the union do indeed contain the property but at least one of them has the property declared under a stricter context than the current one

(It probably should in the second case, but that hasn't been discussed)

### CODE REVIEW

File: tests/baselines/reference/propertyAccess3.errors.txt [hunk 1]
```diff
@@ -1,8 +1,10 @@
...
 
 
 ==== tests/cases/compiler/propertyAccess3.ts (1 errors) ====
     var foo: boolean;
     foo.toBAZ();
         ~~~~~
-!!! error TS2339: Property 'toBAZ' does not exist on type 'boolean'.
\ No newline at end of file
+!!! error TS2339: Property 'toBAZ' does not exist on type 'boolean'.
+!!! error TS2339:   Property 'toBAZ' does not exist on type 'true'.
```
Comment (RyanCavanaugh) [inline tests/baselines/reference/propertyAccess3.errors.txt]: This looks weird - let's exclude `TypeFlags.Boolean` along with `TypeFlags.Enum`

File: src/compiler/checker.ts [hunk 2]
```diff
@@ -10810,6 +10810,20 @@ namespace ts {
                 return propType;
             }
             return getFlowTypeOfReference(node, propType, /*assumeInitialized*/ true, /*flowContainer*/ undefined);
+
+            function reportNonexistantProperty(propNode: Identifier, containingType: Type) {
```
Comment (RyanCavanaugh) [inline src/compiler/checker.ts]: Spelling: Nonexistent

File: tests/baselines/reference/propertyAccess3.errors.txt (hunk 1 above)
Comment (DanielRosenwasser) [inline tests/baselines/reference/propertyAccess3.errors.txt]: Eugh.

How about for enums and booleans we don't dive in? Specifically, you can check `TypeFlags.Boolean` and `TypeFlags.Enum`.

File: tests/baselines/reference/unionPropertyExistance.errors.txt [hunk 3]
```diff
@@ -0,0 +1,60 @@
+tests/cases/compiler/unionPropertyExistance.ts(24,4): error TS2339: Property 'onlyInB' does not exist on type 'AB'.
```
Comment (DanielRosenwasser) [inline tests/baselines/reference/unionPropertyExistance.errors.txt]: Can you rename the file `unionPropertyExistence` (notice the **e**)

File: tests/baselines/reference/propertyAccess3.errors.txt (hunk 1 above)
Comment (RyanCavanaugh) [inline tests/baselines/reference/propertyAccess3.errors.txt]: Great minds think and read GitHub on Sunday nights alike

Comment (DanielRosenwasser) [inline tests/baselines/reference/propertyAccess3.errors.txt]: Actually, `TypeFlags.Primitive` is probably fine to check too.

File: src/compiler/checker.ts (hunk 2 above)
Comment (DanielRosenwasser) [inline src/compiler/checker.ts]: `reportNonexistentProperty`

**Comment by RyanCavanaugh** [comment]
Looks pretty good to me after a couple minor fixes. @mhegazy thoughts on letting this into `master` vs the 2.1 branch?

File: tests/baselines/reference/propertyAccess3.errors.txt (hunk 1 above)
Comment (DanielRosenwasser) [inline tests/baselines/reference/propertyAccess3.errors.txt]: Wow GitHub syncs weird-like

File: tests/baselines/reference/unionPropertyExistance.errors.txt (hunk 3 above)
Comment (oijazsh) [inline tests/baselines/reference/unionPropertyExistance.errors.txt]: Will be happy to.

File: tests/baselines/reference/propertyAccess3.errors.txt (hunk 1 above)
Comment (RyanCavanaugh) [inline tests/baselines/reference/propertyAccess3.errors.txt]: `Primitive` is probably correct. Let's have a test for
```ts
declare const x: "foo" | "bar";
x.nope();
```

**Comment by oijazsh** [comment]
Fixed the typo and excluded unions that are also `Primitive`.

**Comment by mhegazy** [comment]
sorry for the delay. loos good.

**Pull request merged**